src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -19,94 +19,224 @@
+            this.fieldsAndWeights.putAll(fields);
+        }
+
+        this.analyzer = analyzer;
+        this.queryName = queryName;
+        this.defaultOperator = operator;
+        this.minimumShouldMatch = minimumShouldMatch;
+        this.flags = flags;
+        this.settings = new Settings(locale, lowercaseExpandedTerms, lenient, analyzeWildcard);
     }
 
     /**
-     * Add a field to run the query against
+     * Construct a new simple query string query.
+     * 
+     * @param queryText Text to parse query from.
+     * @param boost Boost to apply to resulting Lucene query.
+     * @param fields Fields to run query against.
+     * @param analyzer Analyzer to use for this query.
+     * @param queryName Name of the query.
+     * @param operator Default operator to link boolean clauses, OR otherwise.
+     * @param minimumShouldMatch Optional minimumShouldMatch to apply to this query should it result in a boolean query.
+     * @param flags Flags to apply, ALL by default.
+     * @param locale Locale to use, ROOT by default.
+     * @param lowercaseExpandedTerms Indicate whether parsed terms should be converted to lowercase, true by default.
+     * @param lenient Should parsing be lenient, false by default.
+     * @param analyzeWildcard false by default.
      */
+    public SimpleQueryStringBuilder(String text, float boost, Map<String, Float> fields, String analyzer, String queryName, Operator operator,
+            String minimumShouldMatch, Locale locale, Boolean lowercaseExpandedTerms, Boolean lenient, Boolean analyzeWildcard,
+            SimpleQueryStringFlag... flags) {
+        Preconditions.checkNotNull(text);
+        this.queryText = text;
+        this.boost = boost;
+        this.fieldsAndWeights.putAll(fields);
+        this.analyzer = analyzer;
+        this.queryName = queryName;
+        this.defaultOperator = operator;
+        this.minimumShouldMatch = minimumShouldMatch;
+        int value = 0;
+        if (flags == null || flags.length == 0) {
+            value = SimpleQueryStringFlag.ALL.value;
+        } else {
+            for (SimpleQueryStringFlag flag : flags) {
+                value |= flag.value;
+            }
+        }
+        this.flags = value;
+
+        this.settings = new Settings(locale, lowercaseExpandedTerms, lenient, analyzeWildcard);
+    }
+
+    /** Returns the text to parse the query from. */
+    public String text() {
+        return this.queryText;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public SimpleQueryStringBuilder boost(float boost) {
+        this.boost = boost;
+        return this;
+    }
+    
+    /** Returns the boost to apply to resulting Lucene query.*/
+    public float boost() {
+        return this.boost;
+    }
+
+    /** Add a field to run the query against. */
     public SimpleQueryStringBuilder field(String field) {
-        this.fields.put(field, null);
+        if (field != null && field != "") {
```

Comment:
if we have this check here, shouldn't we have it everywhere else too? e.g. field(String field, float boost) ?